src/main/java/Duke.java

```diff
@@ -1,10 +1,106 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/** Describes the main class. */
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Returns a Duke object that takes in a file path.
+     *
+     * @param path The path to the storage file
+     */
+    public Duke(String path) {
+        storage = new Storage(path);
+        ui = new Ui("Sonia");
+
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (Exception e) {
+            tasks = new TaskList(new ArrayList<>());
+        }
+    }
+
+    /**
+     * Executes the given command.
+     *
+     * @param c The command object
+     */
+    public void executeCommand(Command c) {
```

Comment:
I personally think this method can be slightly too long and has too many responsibilities. In order to follow SLAP closely, maybe you can consider extract the methods to check invalid inputs, perform the task, and print feedback for the task.